frame/support/procedural/src/pallet/parse/mod.rs

```diff
@@ -123,8 +125,31 @@ impl Def {
 					origin = Some(origin::OriginDef::try_from(index, item)?),
 				Some(PalletAttr::Inherent(_)) if inherent.is_none() =>
 					inherent = Some(inherent::InherentDef::try_from(index, item)?),
-				Some(PalletAttr::Storage(span)) =>
-					storages.push(storage::StorageDef::try_from(span, index, item)?),
+				Some(PalletAttr::Storage(span)) => {
+					let mut storage_def = storage::StorageDef::try_from(span, index, item)?;
+					// check for #[benchmarking(cached)] calls
+					if let syn::Item::Type(typ) = item {
+						for attr in typ.attrs.as_slice() {
+							if attr.style == AttrStyle::Outer {
+								if let Some(seg) = attr.path.segments.last() {
+									if let Ok(_) =
+										syn::parse2::<keyword::benchmarking>(seg.to_token_stream())
+									{
+										if let Ok(group) = syn::parse2::<Group>(attr.tokens.clone())
+										{
+											if let Ok(_) =
+												syn::parse2::<keyword::cached>(group.stream())
+											{
+												storage_def.benchmarking_cached = true;
```

Comment:
Why do we parse this here?

Why isn't that done around here: https://github.com/paritytech/substrate/blob/dd1f1b5f291aa2325eddf5f26037b84cfbbe0957/frame/support/procedural/src/pallet/parse/storage.rs#L674 ?

We already parse there the other `pallet::whatever` attributes. Yes, here it would be `benchmarking(cached)`, but it should still be done there.

Reply:
I'll happily rework it there if we are OK with doing something like `#[pallet::whitelist]` instead. Whole reason this was awkward is all the stuff in `storage.rs` is geared around `pallet::` attributes

Reply:
I have refactored things a bit so that the parsing is pulled out into its own function. The reason I couldn't do it where you are saying is that for whatever reason the `benchmarking(cached)` attributes weren't showing up in that `attrs` collection when I tried it from here. But yes this is where I tried parsing first.